api/rest/coordinator.go

```diff
@@ -0,0 +1,231 @@
+package rest
+
+import (
+	"archive/zip"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+const numberOfWorkers = 10
+
+type BundleStatus struct {
+	ID   string
+	node node
+	done bool
+	err  error
+}
+
+// Coordinator is an interface to coordinate the creation of diagnostics bundles
+// across a cluster of nodes
+type Coordinator interface {
+	Create(ctx context.Context, id string, nodes []node) <-chan BundleStatus
```

Comment:
I think this interface should use nodes in the same way as Client so it should be just `host:port` string not a struct
```suggestion
	Create(ctx context.Context, id string, nodes []string) <-chan BundleStatus
```

Reply:
I intentionally didn't do that pretty much just for logging purposes. Otherwise every node would be referred to as `http(s)://<ip>:<port>` which seems to be a fair amount of unnecessary information just to figure out which node the line is referring to though that also means we're passing around IP and role information that isn't functionally necessary.

I don't have strong feelings about this so I'm fine with changing it.

Reply:
In looking at fixing the temporary files, I think we have a decent reason to keep using `node` instead of `string`. Currently bundles name the node directories `<ip>_<role>` which is probably something we'll want to keep so it's easy to see what role the node has. This would also solve the need for generating temp filenames because we now can have a deterministic and unique filename for each zip in the working directory of the bundle. So I think we should stick with `node` and do that. What do you think?

Reply:
Yeah, that's a good point